schema/object.go

```diff
@@ -684,6 +652,40 @@ func (a *AnyTypedObject[T]) Any() TypedObject[any] {
 	return a
 }
 
+// ConvertToObjectSchema attempts to cast the given type or data to an
+// ObjectSchema. If successful it returns true, else it returns false.
```

Comment:
I'd expand this to say that it also extras the object from a ref, and the root from a scope.

Reply:
did it